### PR TITLE
airoha: sync MFD implementation with upstream

### PR DESCRIPTION
**Pull request by Ansuel**

### CODE REVIEW

File: target/linux/airoha/dts/en7581.dtsi [hunk 1]
```diff
@@ -274,24 +274,20 @@
 			#thermal-sensor-cells = <0>;
 		};
 
-		pio: pinctrl {
-			compatible = "airoha,en7581-pinctrl", "syscon";
+		pio: mfd@1fbf0200 {
```
Comment (LorenzoBianconi) [inline target/linux/airoha/dts/en7581.dtsi:277]: @Ansuel is there a better name than pio?

File: target/linux/airoha/files/drivers/mfd/airoha-en7581-mfd.c [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		return PTR_ERR(mfd->base);
+
+	irq = platform_get_irq(pdev, 0);
+	if (irq < 0)
+		return irq;
+
+	airoha_mfd_pinctrl_intr[0].start = irq;
+	airoha_mfd_pinctrl_intr[0].end = irq;
+
+	ret = devm_mfd_add_devices(dev, PLATFORM_DEVID_AUTO,
```
Comment (LorenzoBianconi) [inline target/linux/airoha/files/drivers/mfd/airoha-en7581-mfd.c]: @Ansuel nit: just:
`return devm_mfd_add_devices()`

File: target/linux/airoha/files/drivers/mfd/airoha-en7581-mfd.c [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+static int airoha_mfd_probe(struct platform_device *pdev)
+{
+	struct device *dev = &pdev->dev;
+	struct airoha_mfd *mfd;
+	int irq, ret;
+
+	mfd = devm_kzalloc(dev, sizeof(*mfd), GFP_KERNEL);
+	if (!mfd)
+		return -ENOMEM;
+	platform_set_drvdata(pdev, mfd);
```
Comment (LorenzoBianconi) [inline target/linux/airoha/files/drivers/mfd/airoha-en7581-mfd.c]: @Ansuel nit: blank line here

File: target/linux/airoha/files/drivers/pinctrl/pinctrl-airoha.c [hunk 4]
```diff
@@ -2942,24 +2946,13 @@ static int airoha_pinctrl_probe(struct platform_device *pdev)
...
 	/* build gpio-chip */
-	err = airoha_pinctrl_add_gpiochip(pinctrl, pdev);
-	if (err)
-		return err;
-
-	return of_platform_populate(dev->of_node, NULL, NULL, dev);
+	return airoha_pinctrl_add_gpiochip(pinctrl, pdev);
 }
 
-static const struct of_device_id of_airoha_pinctrl_match[] = {
```
Comment (LorenzoBianconi) [inline target/linux/airoha/files/drivers/pinctrl/pinctrl-airoha.c:2950]: @Ansuel do not drop compatible string even if it is not used at the moment, same for pwm

File: target/linux/airoha/files/drivers/pwm/pwm-airoha.c [hunk 5]
```diff
@@ -81,21 +82,30 @@ struct airoha_pwm {
 /* Duty cycle is relative with 255 corresponding to 100% */
 #define DUTY_FULL	255
 
+static u32 airoha_pwm_rmw(struct airoha_pwm *pc, void __iomem *addr,
```
Comment (LorenzoBianconi) [inline target/linux/airoha/files/drivers/pwm/pwm-airoha.c]: @Ansuel why do we need pc pointer?

File: target/linux/airoha/files/drivers/pwm/pwm-airoha.c [hunk 6]
```diff
@@ -419,16 +431,9 @@ static int airoha_pwm_remove(struct platform_device *pdev)
 	return 0;
 }
 
-static const struct of_device_id airoha_pwm_of_match[] = {
```
Comment (LorenzoBianconi) [inline target/linux/airoha/files/drivers/pwm/pwm-airoha.c:428]: @Ansuel so not drop it

File: target/linux/airoha/dts/en7581.dtsi (hunk 1 above)
Comment (Ansuel) [reply target/linux/airoha/dts/en7581.dtsi:277]: maybe mfd?

**Pull request merged**